app/js/ZipManager.js

```diff
@@ -115,12 +118,31 @@ async function generateZip(projectId, zipfile) {
   zipfile.end()
 }
 
-async function exportProject(projectId) {
-  var zipfile = new yazl.ZipFile()
-  generateZip(projectId, zipfile) // generate zip file in background
-  return zipfile.outputStream
-}
+function exportProject(projectId, callback) {
+  tempy.file.task(path => {
+    return new Promise(resolve => {
+      const outputStream = fs.createWriteStream(path)
+
+      const zipfile = new yazl.ZipFile()
 
-module.exports = {
-  exportProject: util.callbackify(exportProject),
+      pipeline(zipfile.outputStream, outputStream, (err) => {
+        if (err) {
+          callback(err)
+        } else {
+          callback(null, path)
+        }
+
+        resolve() // clean up the temp file
+      })
+
+      generateZip(projectId, zipfile).catch(err => {
+        callback(err)
+        resolve() // clean up the temp file
+      })
+    })
+  }).catch(err => {
```

Comment:
Using mkdtemp would also avoid this weird mix of promises and callbacks

Reply:
Done